web/src/pages/home/components/Users/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@apollo/client';
+import React from 'react'
+import { LIST_USERS } from '../../../../graphql/queries/list-users';
+import { User } from '../../../../typings/user';
+import { UserCard } from '../user-card'
+
+export const GetUsers = () => {
+  const { data, loading, error } = useQuery<{ users: User[] }>(LIST_USERS);
+
+    if (loading) return <>Carregando...</>;
+    if (error || !data?.users) return <>Deu ruim :(</>;
+
+    const { users } = data;
+ 
+    return (
+      <>
+       { users.map(({ id, name, username, website }) => (
+          <UserCard
+            key={id}
+            id={id}
+            name={name}
+            username={username}
+            website={website}
+          />
+        ))
+      }
+      </>
+    )
+  
+}
```

Comment:
Entendo que você fez dessa maneira para conseguir chegar no resultado solicitado a partir de uma renderização condicional, porém você cria alguns problemas com isso.

Cada vez que você muda a renderização do componente acaba tendo que refazer a chamada, se o apollo não tivesse caching out of the box tu faria a mesma requisição várias vezes, o que é ruim para performance.

Além disso, tu cria um problema de semântica no seu código, por que o seu componente tem um nome imperativo **GetUsers**, como se fosse uma função só para pegar usuários, quando na verdade, ele também renderiza o conteúdo.

Para resolver esses problemas, tu poderia ter mantido a lógica para pegar os usuários no componente da página, usado o useLazyQuery, que é o indicado em momentos assim, tu poderia passar o estado que foi definido para a condição de renderização do GetUsers e ter passado a lógica de renderização também para a página e usando esse mesmo estado.

O mesmo pode ser aplicado ao componente de GetPosts.